### ISSUE TITLE
Can't use an undefined value as an ARRAY reference at Hulu.pm line 180

### ISSUE THREAD

**Issue by GoogleCodeExporter** [MiraculousLegendas/get-flash-videos#331]
```
What steps will reproduce the problem?
$ get_flash_videos --debug 
http://www.hulu.com/watch/302739/last-man-standing-guess-whos-coming-to-dinner

What is the expected output? What do you see instead?
1 plugin installed:
- Hulu.pm
Downloading 
http://www.hulu.com/watch/302739/last-man-standing-guess-whos-coming-to-dinner
-> GET 
http://www.hulu.com/watch/302739/last-man-standing-guess-whos-coming-to-dinner
<- 200 text/html; charset=utf-8 (204216)
Trying to open plugin ~/.get_flash_videos/plugins/Www.pm
Trying to open plugin ~/.get_flash_videos/plugins/Hulu.pm
Using method 'hulu' for 
http://www.hulu.com/watch/302739/last-man-standing-guess-whos-coming-to-dinner
Found Hulu CID: 60006542
Returning cached Hulu data
-> GET http://r.hulu.com/videos?content_id=60006542
<- 200 application/xml; charset=utf-8 (8539)
Found Hulu encrypted PID: NO_MORE_RELEASES_PLEASE_60006542
Decrypted Hulu PID: 'NO_MORE_RELEASES_PLEASE_60006542'
Auth: 08fcbe3f1aa3e5d00e48a228a91ec819
Hulu SMIL URL: 
http://s.hulu.com/select?video_id=60006542&v=888324234&ts=1322696271&np=1&vp=1&p
p=Desktop&dp_id=Hulu&region=US&language=en&bcs=cc114b522514b11d752574a1f0f19b43
-> GET 
http://s.hulu.com/select?video_id=60006542&v=888324234&ts=1322696271&np=1&vp=1&p
p=Desktop&dp_id=Hulu&region=US&language=en&bcs=cc114b522514b11d752574a1f0f19b43
<- 200 text/xml (320)
Encrypted XML: 
'e6c69cae4c32d0108c587020098d5f26c0a6cbfe9d82696d5816905a084dc1416bec9d027b062cc
b122fcfbd9b700044a1230258b8b56383c073875e878327b043b22369a40240228ba2e7ebeed3863
015a5ad671dd13123121aade2b8284fd4f982f09e039415de5580aae10a29acc51c50ebb59ced055
76dc349bc0358071a5565c9374fe5041a29dfb5a7a22023746973556f69f6c99423211f113c8321c
e'
XML decrypt key: 
4878B22E76379B55C962B18DDBC188D82299F8F52E3E698D0FAF29A40ED64B21, IV: 
WA7hap7AGUkevuth
XML decrypt key: 
246DB3463FC56FDBAD60148057CB9055A647C13C02C64A5ED4A68F81AE991BF5, IV: 
vyf8PvpfXZPjc7B1
XML decrypt key: 
8CE8829F908C2DFAB8B3407A551CB58EBC19B07F535651A37EBC30DEC33F76A2, IV: 
O3r9EAcyEeWlm5yV
XML decrypt key: 
852AEA267B737642F4AE37F5ADDF7BD93921B65FE0209E47217987468602F337, IV: 
qZRiIfTjIGi3MuJA
XML decrypt key: 
76A9FDA209D4C9DCDFDDD909623D1937F665D0270F4D3F5CA81AD2731996792F, IV: 
d9af949851afde8c
XML decrypt key: 
1F0FF021B7A04B96B4AB84CCFD7480DFA7A972C120554A25970F49B6BADD2F4F, IV: 
tqo8cxuvpqc7irjw
XML decrypt key: 
3484509D6B0B4816A6CFACB117A7F3C842268DF89FCC414F821B291B84B0CA71, IV: 
SUxSFjNUavzKIWSh
XML decrypt key: 
B7F67F4B985240FAB70FF1911FCBB48170F2C86645C0491F9B45DACFC188113F, IV: 
uBFEvpZ00HobdcEo
XML decrypt key: 
40A757F83B2348A7B5F7F41790FDFFA02F72FC8FFD844BA6B28FD5DFD8CFC82F, IV: 
NnemTiVU0UA5jVl0
XML decrypt key: 
d6dac049cc944519806ab9a1b5e29ccfe3e74dabb4fa42598a45c35d20abdd28, IV: 
27b9bedf75ccA2eC
SMIL output: $VAR1 = undef;

Error: Can't use an undefined value as an ARRAY reference at 
/loader/0x96dc2a8/FlashVideo/Site/Hulu.pm line 180.

Couldn't extract Flash movie URL. This site may need specific support adding,
or fixing.

Please confirm the site is using Flash video and if you have Flash available
check that the URL really works(!).

Check for updates by running: /usr/bin/get_flash_videos --update

If the latest version does not support this please open a bug (or
contribute a patch!) at http://code.google.com/p/get-flash-videos/
make sure you include the output with --debug enabled.
Couldn't download any videos.


What version of the product are you using? On what operating system?
get_flash_videos version 1.24 with lastest Hulu.pm plugin
32bit Ubuntu Linux 10.04

Please provide any additional information below.
According to the following post, Hulu just add some additinal parameters to 
their SMIL request:
http://forum.xbmc.org/showpost.php?p=947981&postcount=698
And also from the following post, the "device_id" could be an empty string:
http://forum.xbmc.org/showpost.php?p=948476&postcount=710

BlueCop on XBMC forum had fixed this problem in his branch of the XBMC Hulu 
plugin:
http://code.google.com/p/bluecop-xbmc-repo/source/detail?r=294

Based on his fix, I made a patch (see the attached file) that had been tested 
successfully on my system. The patch just adds the extra parameters to the SMIL 
request by giving an empty "device_id" string. This is kind of a dirty hack, 
but it works so far.

To apply this patch on Linux, download the patch into the 
.get_flash_videos/plugins in your home directory, and then invoke the patch 
command:
$ patch < Hulu.pm.patch
```

Original issue reported on code.google.com by `storm.sf...@gmail.com` on 1 Dec 2011 at 12:06

Attachments:
- [Hulu.pm.patch](https://storage.googleapis.com/google-code-attachments/get-flash-videos/issue-331/comment-0/Hulu.pm.patch)

**Comment by GoogleCodeExporter** [comment]
```
Thanks a million - I have verified. After this patch this works great on OS X 
Lion 10.7.2 with latest getflashvideos 1.24 and rtmpdump 2.4.
```

Original comment by `harperfa...@gmail.com` on 1 Dec 2011 at 2:08

**Comment by GoogleCodeExporter** [comment]
```
Works on my Fedora 16 instance too. Thanks
```

Original comment by `pcwal...@comcast.net` on 1 Dec 2011 at 3:37

**Comment by GoogleCodeExporter** [comment]
```
This works for me too.  Ubuntu 10.4.
```

Original comment by `jsha...@gmail.com` on 1 Dec 2011 at 6:45

**Comment by GoogleCodeExporter** [comment]
```
How do I apply the patch in Windows?
```

Original comment by `Breuste...@gmail.com` on 1 Dec 2011 at 8:17

**Comment by GoogleCodeExporter** [comment]
```
@Breuste:

I have never used get-flash-videos on Win32, so I'm not sure where the Hulu.pm 
will be located. You could try to do a file searching (hotkey: Win-F) to find 
the correct location. Once you find it, save the patch file to that location, 
open a command prompt, and apply the patch using the "patch" windows porting by 
GnuWin32 project:

http://gnuwin32.sourceforge.net/packages/patch.htm

The command line should be pretty much the same with that of Linux.
```

Original comment by `storm.sf...@gmail.com` on 1 Dec 2011 at 2:59

**Comment by GoogleCodeExporter** [comment]
```
Fixed in git on gitorious - 83976f4a. Run get_flash_videos --update to get the 
latest version. Thanks to storm for the patch.
```

Original comment by `zakflash...@gmail.com` on 3 Dec 2011 at 8:30
- Changed state: **Verified**
